system/modules/core/library/Contao/Controller.php

```diff
@@ -1075,14 +1075,23 @@ public static function redirect($strLocation, $intStatus=303)
 	 * @param array   $arrRow       An array of page parameters
 	 * @param string  $strParams    An optional string of URL parameters
 	 * @param string  $strForceLang Force a certain language
-	 * @param boolean $blnFixDomain Check the domain of the target page and append it if necessary
 	 *
 	 * @return string An URL that can be used in the front end
 	 */
-	public static function generateFrontendUrl(array $arrRow, $strParams=null, $strForceLang=null, $blnFixDomain=false)
+	public static function generateFrontendUrl(array $arrRow, $strParams=null, $strForceLang=null)
 	{
 		$strUrl = '';
 
+		if (func_num_args() > 2)
+		{
+			@trigger_error('Using Controller::generateFrontendUrl() with more than two arguments has been deprecated and will no longer work in Contao 5.0.', E_USER_DEPRECATED);
+		}
+
+		if (!isset($arrRow['rootId']))
+		{
+			$arrRow = \PageModel::findWithDetails($arrRow['id'])->row();
```

Comment:
I think we should only fetch the domain (and maybe language) here. We don't know if the alias in the array is actually the same as the given page ID…